### PR TITLE
Intervallisttools

### PR DESCRIPTION
**Pull request by praveenraj2018**
Added the intervallisttools module of GATK. It has been tested locally

### CODE REVIEW

File: software/gatk4/intervallisttools/meta.yml [hunk 1]
```diff
@@ -0,0 +1,44 @@
+name: gatk4_intervallisttools
+
+description: Splits the interval list file into unique, equally-sized interval files and place it under a directory
+keywords:
+  - sort
```
Comment (maxulysse) [inline software/gatk4/intervallisttools/meta.yml:5]:
```suggestion
  - sort
  - intervals
```

File: tests/software/gatk4/intervallisttools/test.yml [hunk 2]
```diff
@@ -0,0 +1,19 @@
+- name: gatk4 intervallisttools test_gatk4_intervallisttools_all
+  command: nextflow run tests/software/gatk4/intervallisttools -entry test_gatk4_intervallisttools_all -c tests/config/nextflow.config
+  tags:
+    - gatk4_intervallisttools_all
+    - gatk4_intervallisttools
```
Comment (maxulysse) [inline tests/software/gatk4/intervallisttools/test.yml:5]:
```suggestion
    - gatk4/intervallisttools
```

File: tests/software/gatk4/intervallisttools/test.yml [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+    - path: output/intervallisttools/test.allparams_split/temp_0002_of_6/2scattered.interval_list
+      md5sum: ba544ea1e99aeaf4b64f61bd729a18fb
+    - path: output/intervallisttools/test.allparams_split/temp_0003_of_6/3scattered.interval_list
+      md5sum: 59d473169711425efaca7810b5093ee5
+    - path: output/intervallisttools/test.allparams_split/temp_0004_of_6/4scattered.interval_list
+      md5sum: 6be1d4371fa7d660c1d2d737e2defafd
+    - path: output/intervallisttools/test.allparams_split/temp_0005_of_6/5scattered.interval_list
+      md5sum: 0c1bb40ec68dccff59d728ac12f05266
+    - path: output/intervallisttools/test.allparams_split/temp_0006_of_6/6scattered.interval_list
+      md5sum: ee7595c84d15eca40326cb7ae418695a
```
[the remark this answers is not in the document]
Comment (praveenraj2018) [reply tests/software/gatk4/intervallisttools/test.yml:18]: output is updated. You can see the number-prefixed to the file name - e.g. 1scattered.interval_list

[the remark this answers is not in the document]
Comment (maxulysse) [reply tests/software/gatk4/intervallisttools/test.yml:18]: Oh I see.

File: tests/software/gatk4/bedtointervallist/test.yml [hunk 4]
```diff
@@ -1,8 +1,8 @@
 - name: gatk4 bedtointervallist test_gatk4_bedtointervallist
   command: nextflow run tests/software/gatk4/bedtointervallist -entry test_gatk4_bedtointervallist -c tests/config/nextflow.config
   tags:
+    - gatk4_bedtointervallist
```
[the remark this answers is not in the document]
Comment (praveenraj2018) [reply tests/software/gatk4/bedtointervallist/test.yml]: Can you check this now? Changed it to "gatk4/bedtointervallist"

[the remark this answers is not in the document]
Comment (maxulysse) [reply tests/software/gatk4/bedtointervallist/test.yml]: Solved

File: software/gatk4/intervallisttools/meta.yml [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+  - intervals
```
Comment (maxulysse) [inline software/gatk4/intervallisttools/meta.yml:6]:
```suggestion
  - bed
  - interval list
```

**Comment by maxulysse** [comment]
@KevinMenden any idea why the linting fails on the module?

**Comment by KevinMenden** [comment]
Hmm it says it cant find the module :thinking: I'll clone the code and test locally

File: software/gatk4/intervallisttools/main.nf [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+
+params.options = [:]
+options        = initOptions(params.options)
+
+process GATK4_INTERVALLISTTOOLS {
+    tag "$meta.id"
+    label 'process_medium'
+    publishDir "${params.outdir}",
+        mode: params.publish_dir_mode,
+        saveAs: { filename -> saveFiles(filename:filename, options:params.options, publish_dir:getSoftwareName(task.process), publish_id:meta.id) }
```
Comment (KevinMenden) [inline software/gatk4/intervallisttools/main.nf:12]:
```suggestion
        saveAs: { filename -> saveFiles(filename:filename, options:params.options, publish_dir:getSoftwareName(task.process), meta:meta, publish_by_meta:['id']) }
```

that was changed a while ago

File: tests/software/gatk4/intervallisttools/test.yml [hunk 7]
```diff
@@ -0,0 +1,17 @@
+- name: gatk4 intervallisttools test_gatk4_intervallisttools_all
+  command: nextflow run tests/software/gatk4/intervallisttools -entry test_gatk4_intervallisttools_all -c tests/config/nextflow.config
+  tags:
+    - gatk4
+    - gatk4_intervallisttools
+    - gatk4_intervallisttools_all
```
Comment (KevinMenden) [inline tests/software/gatk4/intervallisttools/test.yml:6]:
```suggestion
    - gatk4
    - gatk4/intervallisttools
```

new tag guideline

File: tests/config/pytest_software.yml [hunk 8]
```diff
@@ -290,6 +290,10 @@ gatk4/variantfiltration:
   - software/gatk4/variantfiltration/**
   - tests/software/gatk4/variantfiltration/**
 
+gatk4_intervallisttools:
```
Comment (KevinMenden) [inline tests/config/pytest_software.yml:293]:
```suggestion
gatk4/intervallisttools:
```
that bad boy was causing the trouble

File: tests/software/gatk4/intervallisttools/test.yml [hunk 9]
```diff
@@ -0,0 +1,16 @@
+- name: gatk4 intervallisttools test_gatk4_intervallisttools_all
```
Comment (maxulysse) [inline tests/software/gatk4/intervallisttools/test.yml:1]:
```suggestion
- name: gatk4 intervallisttools test_gatk4_intervallisttools
```

File: tests/software/gatk4/intervallisttools/test.yml [hunk 10]
```diff
@@ -0,0 +1,16 @@
...
+  command: nextflow run tests/software/gatk4/intervallisttools -entry test_gatk4_intervallisttools_all -c tests/config/nextflow.config
```
Comment (maxulysse) [inline tests/software/gatk4/intervallisttools/test.yml:2]:
```suggestion
  command: nextflow run tests/software/gatk4/intervallisttools -entry test_gatk4_intervallisttools -c tests/config/nextflow.config
```

**Comment by maxulysse** [comment]
@KevinMenden I think we got it this time

**Review by KevinMenden — APPROVED** [review]

**Comment by maxulysse** [comment]
> @KevinMenden I think we got it this time

Famous last words before failure

**Comment by KevinMenden** [comment]
True, now we broke something else :facepalm: but shouldn't be too bad, just something with the tests :shrug:

File: tests/software/gatk4/intervallisttools/test.yml [hunk 11]
```diff
@@ -0,0 +1,16 @@
+- name: gatk4 intervallisttools test_gatk4_intervallisttools
+  command: nextflow run tests/software/gatk4/intervallisttools -entry test_gatk4_intervallisttools -c tests/config/nextflow.config
+  tags:
+    - gatk4
+    - gatk4/intervallisttools
```
Comment (maxulysse) [inline tests/software/gatk4/intervallisttools/test.yml:5]:
```suggestion
    - gatk4_intervallisttools
```

File: tests/software/gatk4/intervallisttools/main.nf [hunk 12]
```diff
@@ -0,0 +1,16 @@
+#!/usr/bin/env nextflow
+
+nextflow.enable.dsl = 2
+
+test_options = ['args': '--SCATTER_COUNT 6 --SUBDIVISION_MODE BALANCING_WITHOUT_INTERVAL_SUBDIVISION_WITH_OVERFLOW --UNIQUE true --SORT true']
+include { GATK4_BEDTOINTERVALLIST } from '../../../../software/gatk4/bedtointervallist/main.nf' addParams( options: [:] )
+include { GATK4_INTERVALLISTTOOLS as INTERVALLISTTOOLS_ALL } from '../../../../software/gatk4/intervallisttools/main.nf' addParams( options: test_options )
+
+workflow test_gatk4_intervallisttools_all {
```
Comment (maxulysse) [inline tests/software/gatk4/intervallisttools/main.nf:9]:
```suggestion
workflow test_gatk4_intervallisttools {
```

File: tests/software/gatk4/intervallisttools/main.nf [hunk 13]
```diff
@@ -0,0 +1,16 @@
...
+
+    input = [ [ id:'test' ], [ file(params.test_data['sarscov2']['genome']['test_bed'], checkIfExists: true) ]]
+    dict = file(params.test_data['sarscov2']['genome']['genome_dict'], checkIfExists: true)
+
+    GATK4_BEDTOINTERVALLIST ( input, dict )
+    INTERVALLISTTOOLS_ALL ( GATK4_BEDTOINTERVALLIST.out.interval_list )
```
Comment (maxulysse) [inline tests/software/gatk4/intervallisttools/main.nf:15]:
```suggestion
    INTERVALLISTTOOLS ( GATK4_BEDTOINTERVALLIST.out.interval_list )
```

**Review by maxulysse — APPROVED** [review]
LGTM

@praveenraj2018 you can hit squash and merge

**Pull request merged**